docs/git-guide.md

```diff
@@ -0,0 +1,1052 @@
+
+### Step 1a: Create your fork and clone to your machine
+
+The following steps you'll only need to do the first time you setup a machine for contributing to a given Zulip project. You'll need to repeat the steps for any additional Zulip projects ([list][github-zulip]) that you work on.
+
+The first thing you'll want to do to contribute to Zulip is fork ([see how][github-help-fork]) the appropriate [zulip repository][github-zulip]. For the main server app, this is [zulip/zulip][github-zulip-zulip].
+
+Next, clone your fork to your local machine:
+
+```
+$ git clone https://github.com/YOUR_USERNAME/zulip.git
+Cloning into 'zulip'
+remote: Counting objects: 86768, done.
+remote: Compressing objects: 100% (15/15), done.
+remote: Total 86768 (delta 5), reused 1 (delta 1), pack-reused 86752
+Receiving objects: 100% (86768/86768), 112.96 MiB | 523.00 KiB/s, done.
+Resolving deltas: 100% (61106/61106), done.
+Checking connectivity... done.
+```
+
+### Step 1b: Connect your fork to Zulip upstream
+
+Next you'll want to [configure an upstream remote repository][github-help-conf-remote] for your fork of Zulip. This will allow you to [sync changes][github-help-sync-fork] from the main project back into your fork.
+
+First, show the currently configured remote repository:
+
+```
+$ git remote -v
+origin  git@github.com:YOUR_USERNAME/zulip.git (fetch)
+origin  git@github.com:YOUR_USERNAME/zulip.git (push)
+```
+
+Note: If you've cloned the repository using a graphical client, you may already have the upstream remote repository configured. For example, when you clone [zulip/zulip][github-zulip-zulip] with the GitHub desktop client it configures the remote repository `zulip` and you see the following output from `git remote -v`:
+
+```
+origin   https://github.com/koehlerwtf/zulip.git (fetch)
+origin   https://github.com/koehlerwtf/zulip.git (push)
+zulip    https://github.com/zulip/zulip.git (fetch)
+zulip    https://github.com/zulip/zulip.git (push)
+```
+
+If your client hasn't automatically configured a remote for zulip/zulip, you'll need to with:
+
+```
+$ git remote add upstream https://github.com/zulip/zulip.git
+```
+
+Finally, confirm that the new remoate repository, upstream, has been configured:
+
+```
+$ git remote -v
+origin  git@github.com:YOUR_USERNAME/zulip.git (fetch)
+origin  git@github.com:YOUR_USERNAME/zulip.git (push)
+upstream https://github.com/zulip/zulip.git (fetch)
+upstream https://github.com/zulip/zulip.git (push)
+```
+
+### Step 2: Setup the Zulip dev environment
+
+If you haven't already, now is a good time to install the Zulip dev environment ([overview][zulip-rtd-dev-overview]). If you're new to working on Zulip or open source projects in general, we recommend following our [detailed guide for first-time contributors][zulip-rtd-dev-first-time].
+
+### Step 3: Configure Travis CI (continuos integration)
```

Comment:
That's going to make the entry in the ToC super long and less readable (at least for me). I'd rather keep it short unless you feel strongly about the longer version?